tests_e2e/tests/scripts/agent_update-self_update_test_setup

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script prepares the new agent and install it on the vm
+#
+
+set -euo pipefail
+
+usage() (
+    echo "Usage: agent_update-self_update_test_setup -p|--package <path> -v|--version <version>"
+    exit 1
+)
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -p|--package)
+            shift
+            if [ "$#" -lt 1 ]; then
+                usage
+            fi
+            package=$1
+            shift
+            ;;
+        -v|--version)
+            shift
+            if [ "$#" -lt 1 ]; then
+                usage
+            fi
+            version=$1
+            shift
+            ;;
+        *)
+            usage
+    esac
+done
+if [ "$#" -ne 0 ] || [ -z ${package+x} ] || [ -z ${version+x} ]; then
+    usage
+fi
+
+echo "updating the related to self-update flags"
+update-waagent-conf Debug.EnableGAVersioning=n Debug.SelfUpdateHotfixFrequency=120 Debug.SelfUpdateRegularFrequency=120 Autoupdate.Frequency=120
+agent-service stop
+mv /var/log/waagent.log /var/log/waagent.$(date --iso-8601=seconds).log
+
+echo "Cleaning up the existing agents"
+rm -rf /var/lib/waagent/WALinuxAgent-*
+
+echo "Installing $package as version $version..."
+unzip.py $package /var/lib/waagent/WALinuxAgent-$version
+agent-service restart
```

Comment:
Will the published versions > 2.8.9.9 be downloaded during restart? Or would 2.8.9.9 get started, and then download the published versions?

Reply:
I purge the existing versions before step, so that on restart agent start with 2.8.9.9 and then download the latest version if > 2.8.9.9